pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -264,21 +268,38 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 	}
 	objs = append([]*uns.Unstructured{app}, objs...)
 
-	// Set up the Pod reconciler before we start creating DaemonSets/Deployments
+	// Set up the Pod reconciler before we start creating DaemonSets/Deployments/StatefulSets
 	daemonSets := []types.NamespacedName{}
 	deployments := []types.NamespacedName{}
+	statefulSets := []types.NamespacedName{}
 	relatedObjects := []configv1.ObjectReference{}
+	relatedClusterObjects := []network.RelatedObject{}
 	for _, obj := range objs {
 		if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "DaemonSet" {
 			daemonSets = append(daemonSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
 		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "Deployment" {
 			deployments = append(deployments, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "StatefulSet" {
+			statefulSets = append(statefulSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
 		}
 		restMapping, err := r.mapper.RESTMapping(obj.GroupVersionKind().GroupKind())
 		if err != nil {
 			log.Printf("Failed to get REST mapping for storing related object: %v", err)
 			continue
 		}
+		if obj.GetClusterName() == cnoclient.ManagementClusterName {
```

Comment:
Shouldn't that be `if obj.GetClusterName() != cnoclient.DefaultClusterName`?

Reply:
This is intended, it puts the management cluster objects in a new relatedClusterObjects struct and track them separately.
management cluster objects will be updated in clusterOperator annotation, guest related objects are updated in clusterOperator status.

Reply:
But isn't that true for all non-default-cluster objects? Why would the management cluster be special?

Reply:
@squeed I remembered we have this summary for RelatedObjects that: we should not be adding management cluster objects to RelatedObjects. We might need to add them as an annotation on our ClusterOperator object for internal reasons (cleanup). This is why the management objects are tracked in a different struct in status manager and be updated later in the network.operator annotation.

Reply:
Right, what I'm saying is that this conditional should probably be `obj.GetClusterName() != ""`

Reply:
Done